test/solidity/Facets/ChainflipFacet.t.sol

```diff
@@ -0,0 +1,472 @@
+        // Set source chain to Arbitrum for these tests
+        vm.chainId(CHAIN_ID_ARBITRUM);
+        vm.roll(208460950); // Set specific block number for Arbitrum chain
+
+        vm.startPrank(USER_SENDER);
+
+        // Test Ethereum mapping
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+        bridgeData.destinationChainId = CHAIN_ID_ETHEREUM;
+        initiateBridgeTxWithFacet(false);
+
+        // Test Arbitrum mapping (should fail as same network)
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+        bridgeData.destinationChainId = CHAIN_ID_ARBITRUM;
+        vm.expectRevert(CannotBridgeToSameNetwork.selector);
+        initiateBridgeTxWithFacet(false);
+
+        // Test Solana mapping
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+        bridgeData.destinationChainId = CHAIN_ID_SOLANA;
+        bridgeData.receiver = LibAsset.NON_EVM_ADDRESS;
+        validChainflipData.nonEVMReceiver = bytes32(
+            abi.encodePacked("EoW7FWTdPdZKpd3WAhH98c2HMGHsdh5yhzzEtk1u68Bb")
+        );
+        initiateBridgeTxWithFacet(false);
+
+        // Test Bitcoin mapping
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+        bridgeData.destinationChainId = CHAIN_ID_BITCOIN;
+        validChainflipData.nonEVMReceiver = bytes32(
+            abi.encodePacked("bc1q6l08rtj6j907r2een0jqs6l7qnruwyxfshmf8a")
+        );
+        initiateBridgeTxWithFacet(false);
+
+        // Test invalid chain ID
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+        bridgeData.destinationChainId = 137; // Polygon
+        vm.expectRevert(ChainflipFacet.UnsupportedChainflipChainId.selector);
+        initiateBridgeTxWithFacet(false);
+
+        vm.stopPrank();
+    }
+
+    function testRevert_WhenUsingEmptyNonEVMAddress() public {
+        bridgeData.receiver = LibAsset.NON_EVM_ADDRESS;
+        bridgeData.destinationChainId = CHAIN_ID_SOLANA;
+        validChainflipData = ChainflipFacet.ChainflipData({
+            dstToken: 6,
+            nonEVMReceiver: bytes32(0), // Empty address should fail
+            message: "",
+            gasAmount: 0,
+            cfParameters: ""
+        });
+
+        vm.startPrank(USER_SENDER);
+
+        // approval
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+
+        vm.expectRevert(ChainflipFacet.EmptyNonEvmAddress.selector);
+
+        initiateBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function test_CanBridgeTokensWithDestinationCall()
+        public
+        assertBalanceChange(
+            ADDRESS_USDC,
+            USER_SENDER,
+            -int256(defaultUSDCAmount)
+        )
+        assertBalanceChange(ADDRESS_USDC, USER_RECEIVER, 0)
+        assertBalanceChange(ADDRESS_DAI, USER_SENDER, 0)
+        assertBalanceChange(ADDRESS_DAI, USER_RECEIVER, 0)
+    {
+        // Set destination to Arbitrum where our receiver contract is
```

Comment:
I think in that case we would also set the receiverAddress to the address of the ChainflipReceiver.
Which makes me question if we should not always use a "receiverAddress" in the ChainflipData  so that we can actually implement this case (with destination call) correctly:

- bridgeData should always have the receiver address of the user
- but the facet needs to use the address of our receiver

I dont think we currently do/support this, correct?